src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/FlowAnalysis/GlobalFlowStateDictionaryFlowOperationVisitor.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.GlobalFlowStateAnalysis;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines.AvoidMultipleEnumerations.FlowAnalysis
+{
+    using GlobalFlowStateDictionaryAnalysisData = DictionaryAnalysisData<AnalysisEntity, GlobalFlowStateDictionaryAnalysisValue>;
+    using GlobalFlowStateDictionaryAnalysisResult = DataFlowAnalysisResult<GlobalFlowStateDictionaryBlockAnalysisResult, GlobalFlowStateDictionaryAnalysisValue>;
+
+    internal abstract class GlobalFlowStateDictionaryFlowOperationVisitor : GlobalFlowStateDataFlowOperationVisitor<
+        GlobalFlowStateDictionaryAnalysisContext,
+        GlobalFlowStateDictionaryAnalysisResult,
+        GlobalFlowStateDictionaryAnalysisValue>
+    {
+        protected GlobalFlowStateDictionaryFlowOperationVisitor(
+            GlobalFlowStateDictionaryAnalysisContext analysisContext) : base(analysisContext, true)
+        {
+        }
+
+        public override GlobalFlowStateDictionaryAnalysisData GetEmptyAnalysisData()
+            => new();
+
+        public sealed override (GlobalFlowStateDictionaryAnalysisData output, bool isFeasibleBranch) FlowBranch(BasicBlock fromBlock, BranchWithInfo branch, GlobalFlowStateDictionaryAnalysisData input)
+        {
+            EnsureInitialized(input);
+            return base.FlowBranch(fromBlock, branch, input);
+        }
+
+        protected override bool Equals(GlobalFlowStateDictionaryAnalysisData value1, GlobalFlowStateDictionaryAnalysisData value2)
+            => GlobalFlowStateDictionaryAnalysis.Domain.Equals(value1, value2);
+
+        protected override GlobalFlowStateDictionaryAnalysisValue GetAbstractDefaultValue(ITypeSymbol type)
+            => GlobalFlowStateDictionaryAnalysisValue.Empty;
+
+        protected override GlobalFlowStateDictionaryAnalysisData GetClonedAnalysisData(GlobalFlowStateDictionaryAnalysisData analysisData)
```

Comment:
Same here, can't this be moved to the base type?

Reply:
Also for other methods below that are not using `GlobalFlowStateDictionaryAnalysis.Domain`

Reply:
For this method I have moved it to base type. I will comment in the all the follow methods for the reason why it can't be moved